src/IW/Sync/Search.hs

```diff
@@ -1,31 +1,60 @@
-module IW.Server.Search where
+module IW.Sync.Search
+       ( fetchAllHaskellRepos
+       , fetchAllHaskellIssues
+       , fetchOpenIssuesWithLabels
+       ) where
 
-import GitHub (Error, Repo, SearchResult, Issue)
+import GitHub (SearchResult (..))        
 import GitHub.Endpoints.Search (searchRepos, searchIssues)
 
+import IW.App (WithError)
+import IW.Db (WithDb)
+import IW.App.Error (AppErrorType, throwError)
+
+import qualified GitHub (Repo, Issue, Error (..))
+
+
 -- | Fetch all repositories with Haskell language
-fetchAllHaskellRepos :: IO (Either Error (SearchResult Repo))
+fetchAllHaskellRepos :: IO (Either GitHub.Error (SearchResult GitHub.Repo))
```

Comment:
Our ultimate goal is to have function:

```haskell
fetchAllHaskellRepos :: ... m => m [Repo]
```

So you can keep existing functions and then implement wrappers around them that work nicer with our interface.

Reply:
Nice